hypervisor/arch/x86/guest/vmsr.c

```diff
@@ -51,6 +51,43 @@ static void enable_msr_interception(uint8_t *bitmap, uint32_t msr_arg)
 	write_map[(msr >> 3U)] = value;
 }
 
+/*
+ * Enable read and write msr interception for x2APIC MSRs
+ * MSRs that are not supported in the x2APIC range of MSRs,
+ * i.e. anything other than the ones below and between
+ * 0x802 and 0x83F, are not intercepted
+ */
+
+static void intercept_x2apic_msrs(uint8_t *msr_bitmap_arg)
+{
+	uint8_t *msr_bitmap = msr_bitmap_arg;
```

Comment:
Please use a static array to contain all the MSRs we want to intercept, and simplify this API to loop the array.

BTW, can we add those MSRs to emulated_msrs?

Reply:
addressed by adding a static array of x2apic MSRs